File: go/src/day2/puzzle.go
package day2

func SolvePart1(input [][]int) int {
	checksum := 0

	for _, rowOfNumbers := range input {
		min := min(rowOfNumbers)
		max := max(rowOfNumbers)

		checksum += max - min
	}

	return checksum
}

func SolvePart2(input [][]int) int {
	checksum := 0

	for _, rowOfNumbers := range input {
		for i := 0; i < len(rowOfNumbers); i++ {
			for j := 0; j < len(rowOfNumbers); j++ {
				if i != j && rowOfNumbers[j] != 0 && rowOfNumbers[i]%rowOfNumbers[j] == 0 {
					checksum += rowOfNumbers[i] / rowOfNumbers[j]
				}
			}
		}
	}

	return checksum
}

func min(numbers []int) int {
	min := numbers[0]

	for _, number := range numbers {
		if number < min {
			min = number
		}
	}

	return min
}

func max(numbers []int) int {
	max := numbers[0]

	for _, number := range numbers {
		if number > max {
			max = number
		}
	}

	return max
}

File: 2017/dotnet/src/Day2/Puzzle.cs
using System.Linq;

namespace AdventOfCode.Day2
{
    public static class Puzzle
    {
        public static int SolvePart1(int[][] input)
        {
            var checksum = 0;

            foreach (var rowOfNumbers in input)
            {
                var min = rowOfNumbers.Min();
                var max = rowOfNumbers.Max();

                checksum += max - min;
            }

            return checksum;
        }

        public static int SolvePart2(int[][] input)
        {
            var checksum = 0;

            foreach (var rowOfNumbers in input)
            {
                for (int i = 0; i < rowOfNumbers.Length; i++)
                for (int j = 0; j < rowOfNumbers.Length; j++)
                {
                    if (i != j &&
                        rowOfNumbers[j] != 0 &&
                        rowOfNumbers[i] % rowOfNumbers[j] == 0)
                    {
                        checksum += rowOfNumbers[i] / rowOfNumbers[j];
                    }
                }
            }

            return checksum;
        }
    }
}
